web/packages/teleport/src/Notifications/Notification.tsx

```diff
@@ -0,0 +1,328 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { formatDistanceToNowStrict } from 'date-fns';
+
+import * as Icons from 'design/Icon';
+
+import { Notification as NotificationType } from 'teleport/services/notifications';
+
+import { notificationContentFactory } from './notificationContentFactory';
+import Text from 'design/Text';
+import { ButtonSecondary } from 'design/Button';
+import { MenuIcon, MenuItem } from 'shared/components/MenuAction';
+import Dialog, {
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from 'design/Dialog';
+
+export function Notification({
+  notification,
+}: {
+  notification: NotificationType;
+}) {
+  const content = notificationContentFactory(notification);
+
+  // Whether to show the text content dialog. This is only ever used for user-created notifications which only contain informational text
+  // and don't redirect to any page.
+  const [showTextContentDialog, setShowTextContentDialog] = useState(false);
+
+  let AccentIcon;
+  switch (content.type) {
+    case 'success':
+    case 'success-alt':
+      AccentIcon = Icons.Check;
+      break;
+    case 'informational':
+      AccentIcon = Icons.Info;
+      break;
+    case 'warning':
+      AccentIcon = Icons.Warning;
+      break;
+    case 'failure':
+      AccentIcon = Icons.Cross;
+      break;
+  }
+
+  const formattedDate = formatDate(notification.createdDate);
+
+  function onMarkAsRead() {
+    // TODO rudream - add mark as read functionality
```

Comment:
Yes, and it will also make a request to the backend to update that notification and mark it as read for the user so that the next time it is fetched, `clicked` will already be true.

Reply:
So what would you say for renaming `clicked` to `read` (or `isRead`)? It would be more consistent.

Reply:
We actually want to stick with the `clicked` naming, because a `clicked` notification is different from a `read` one, this is also how we are naming it in the backend resource.
 
A read (or "seen") notification is just a notification that the user has seen in their notification list, they shouldn't be marked as `clicked`. We never mark individual notifications as read. Whether a user has read a notification or not will do nothing other than determine whether we show the new notifications badge in the toolbar:
<img width="99" alt="image" src="https://github.com/gravitational/teleport/assets/56373201/5617d52c-876b-42da-be32-765ad5f4ec97">

I explained how this will work in [this section of the RFD](https://github.com/gravitational/teleport/blob/master/rfd/0156-notifications.md#user-last-seen-notification-item).

A `clicked` notification is a notification that the user has actually clicked on or used the quick action on, this is when we want to apply the different styles we defined. 

I'll rename this function to `onMarkAsClicked` to be more consistent, but the user will see "mark as read" on the button since that's the more intuitive language they'd expect and what's in the Figma mockups.